sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/models/GroupAllocation.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.feature.management.implementation.models;
+
+import java.util.Map;
+
+public class GroupAllocation {
+
+    private String variant;
+
+    private Map<String, String> groups;
```

Comment:
Why do we need to set `groups` as a map? Should it be a string array?
I'm not sure what is the key of each group.

Reply:
This is a big, long complication due to use using a schema that doesn't match what a Spring library would expect. The work around is that we catch the property changes midway through the process and do the final processing ourselves. But it results in all lists turning into dictionaries where the key of the value is the location it was in the list. I found it easier to just keep the map as is and just used the values when we need to.

I might try looking over this again some other time when we have a moment.

Reply:
Ok, I see. Thanks